packages/web-config-server/src/apiV1/legacyReport.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import ViewRouteHandler from './view';
+import { DashboardItemPermissionChecker } from './permissions';
+
+export default class extends ViewRouteHandler {
+  static PermissionsChecker = DashboardItemPermissionChecker;
+
+  fetchReport = async ({ itemCode, drillDownLevel }) => {
+    const baseDashboardItem = await this.models.dashboardItem.findOne({
+      code: itemCode,
+      legacy: true,
+    });
+    let dashboardItem = baseDashboardItem;
+    if (drillDownLevel) {
+      const drillDownCode = baseDashboardItem.config.drillDown?.itemCodes[drillDownLevel];
```

Comment:
This is a smart way of continuing to support the current front end behaviour! Love the way you've structured it with the inheritance too. 

My plan with drill down support was that the front end should change so that it just requests a different report, based on the drill down item code in the base config. I think that's a little cleaner than having it work the drill down out here while still using the base item code. 

The implication there is a slightly different structure for the migration, where instead of putting the item codes into an array on the base config, each config has just one item code it can drill down to (so the second layer in has the code for the third layer in)

We don't need to put together the front end support for this now, but we probably do need to change the migration and delete this code. What do you think about that? Can you think of any downsides?